config_example.sh

```diff
@@ -550,6 +550,23 @@ set -x
 #
 #export MIRROR_IMAGES=true
 
+# OC_MIRROR -
+# When MIRROR_IMAGES is true, use the 'mirror-registry' command to create the
+# registry and the 'oc mirror' command to do the mirroring.
+# Note, this is currently only supported for the Agent implementation.
+# Default: false
+#
+# When using OC_MIRROR the auths for the mirror must be added to .docker/config.json
+# or an UNAUTHORIZED error will result when attempting to use it. An example entry
+# in this file is:
+# "virthost.ostest.test.metalkube.org:8443": { "auth": "c3RhY2s6cXVheXBhc3N3b3Jk" },
+# where the password used for the mirror is generated by
+# 'echo -n "${QUAY_REGISTRY_USER}:${QUAY_REGISTRY_PASS}" | base64 -w0`'
+#
+# In addition, the authorization for quay.io and registry.redhat.io must be added
+# to .docker/config.json, the same content that is in $PULL_SECRET_FILE.
+#export OC_MIRROR=false
```

Comment:
What happens if this config var is enabled in a non-agent flow?

Reply:
Its a noop - we don't use if for anything outside of the agent

Reply:
It doesn't seem fully correct. An external dev-scripts user, by reading the doc will have not clue about it. 
Also, by setting the `MIRROR_IMAGES`, another local registry will be anyway setup during the configure [step](https://github.com/openshift-metal3/dev-scripts/blob/9f4c324a56d8b43fbb9be01921144f6ccf37e7cd/02_configure_host.sh#L71-L73) in the agent flow, thus resulting in two local registries, ie: 
```
CONTAINER ID  IMAGE                                         COMMAND               CREATED            STATUS                PORTS                   NAMES
33070dd9195c  quay.io/libpod/registry:2.8                   /etc/docker/regis...  About an hour ago  Up About an hour ago                          registry
...
09c4bbd671d8  registry.redhat.io/quay/quay-rhel8:v3.7.8     registry              59 minutes ago     Up 59 minutes ago     0.0.0.0:8443->8443/tcp  quay-app
```
I'd still think that a `battery swap` approach for the registry implementation (default or quay) would be preferred in general and less confusing from my point of view. If not something to be addressed in this PR, I'd at least make it clear that's something temporarly specific for the agent flow (thus the config name)

Reply:
I added a check to not call setup_local_registry() if OC_MIRROR is set. The legacy registry might still be set up from a previous run as I don't think it gets cleaned up, but fixing that is out of scope.

It it would be clearer as I can set MIRROR_IMAGES when OC_MIRROR is set which is what we do with IPv6.